examples/gno.land/p/moul/deque/deque.gno

```diff
@@ -0,0 +1,345 @@
+func (d *Deque) PushFront(items ...any) {
+	for _, item := range items {
+		d.pushFrontSingle(item)
+	}
+}
+
+// pushFrontSingle adds a single element to the front
+func (d *Deque) pushFrontSingle(item any) {
+	node := &Node{value: item}
+
+	if d.head == nil {
+		// First element
+		d.head = node
+		d.tail = node
+	} else {
+		// Link to existing head
+		d.head.prev = node
+		node.next = d.head
+		d.head = node
+	}
+
+	d.size++
+
+	// Handle size limit by removing from back
+	if d.max > 0 && d.size > d.max {
+		d.popBackNode()
+	}
+}
+
+// Size returns the current number of elements
+func (d *Deque) Size() int {
+	return d.size
+}
+
+// MaxSize returns the maximum size limit (0 = unlimited)
+func (d *Deque) MaxSize() int {
+	return d.max
+}
+
+// IsEmpty returns true if the deque is empty
+func (d *Deque) IsEmpty() bool {
+	return d.size == 0
+}
+
+// IsBounded returns true if the deque has a size limit
+func (d *Deque) IsBounded() bool {
+	return d.max > 0
+}
+
+// First returns the front element without removing it.
+// Returns nil if the deque is empty.
+func (d *Deque) First() any {
+	if d.head == nil {
+		return nil
+	}
+	return d.head.value
+}
+
+// Last returns the back element without removing it.
+// Returns nil if the deque is empty.
+func (d *Deque) Last() any {
+	if d.tail == nil {
+		return nil
+	}
+	return d.tail.value
+}
+
+// Get returns the element at the specified index (0 = front).
+// Returns nil if index is out of bounds.
+func (d *Deque) Get(index int) any {
+	if index < 0 || index >= d.size {
+		return nil
+	}
+
+	node := d.head
+	for i := 0; i < index; i++ {
+		node = node.next
+	}
+	return node.value
+}
+
+// List returns all elements as a slice, ordered from front to back
+func (d *Deque) List() []any {
+	if d.size == 0 {
+		return nil
+	}
+
+	result := make([]any, d.size)
+	node := d.head
+	for i := 0; i < d.size; i++ {
+		result[i] = node.value
+		node = node.next
+	}
+	return result
+}
+
+// Enumerate calls fn for each element with its index (0-based from front)
+func (d *Deque) Enumerate(fn func(index int, value any) bool) {
+	node := d.head
+	for i := 0; i < d.size; i++ {
+		if fn(i, node.value) {
+			break
+		}
+		node = node.next
+	}
+}
+
+// Clear removes all elements from the deque
+func (d *Deque) Clear() {
+	d.head = nil
+	d.tail = nil
+	d.size = 0
+}
+
+// SetMaxSize changes the maximum size limit.
+// If the new limit is smaller than current size, elements are removed from front.
+func (d *Deque) SetMaxSize(maxSize int) {
+	d.max = maxSize
+
+	if maxSize > 0 {
+		for d.size > maxSize {
+			d.popFrontNode()
+		}
+	}
+}
+
+// Iterator returns a new iterator for forward traversal
+func (d *Deque) Iterator() *Iterator {
+	return &Iterator{
+		current: d.head,
+	}
+}
+
+// ReverseIterator returns a new iterator for backward traversal
+func (d *Deque) ReverseIterator() *ReverseIterator {
+	return &ReverseIterator{
+		current: d.tail,
+	}
+}
```

Comment:
Maybe would be better to implement these as Go [iterators](https://pkg.go.dev/iter), for forwards/cross compatibility